### ISSUE TITLE
Verifying binary releases

### ISSUE THREAD

**Issue by mstorsjo** [cisco/openh264#893]
At some point there's been said that users of the binary releases will be able to verify those releases by themselves compiling an identical binary from source for comparison. (Can't find the exact quote right now.)

To actually be able to do that, it would be good to know more in detail what compiler/toolchain version that has been used to build the binaries - without that it's essentially impossible to try to recreate an identical binary.

**Comment by maxnordlund** [comment]
What about using some sort of container, such as [docker](https://www.docker.com/) or [rkt](https://github.com/coreos/rkt)? This will ensure a mostly deterministic environment, as long as you don't run `apt-get update` or similar.

**Comment by fluffy** [comment]
I put together a docker build setup for making a linux build. It is on a branch at 

https://github.com/cisco/openh264/tree/cj-build1

It does seem to make deterministic builds. 

I think this is probably a good approach going forward but we need to see some sort of container to make a windows build

**Comment by maxnordlund** [comment]
Yes, that is something we need and maybe http://mxe.cc/ can be used for cross-compiling? Then you don't need to be on windows or have access to its binaries.

That branch is a fair bit behind master ~750 commits. Could you try to update it, please?

**Comment by luser** [comment]
The Tor project has been doing deterministic builds (including of Firefox):
https://blog.torproject.org/blog/deterministic-builds-part-two-technical-details

Using their methodology is probably the best way to go about this.

**Comment by nanonyme** [comment]
One possibility would be to use something like BuildStream to create the binaries inside a sandbox. https://gitlab.com/freedesktop-sdk/freedesktop-sdk produces a reproducible SDK that can be used to build binaries. This project currently runs their own CI inside docker.
